internal/build/tar.go

```diff
@@ -45,7 +45,7 @@ func clearUIDAndGID(h *tar.Header) {
 }
 
 func (a *ArchiveBuilder) archiveDf(ctx context.Context, df dockerfile.Dockerfile) error {
-	span, ctx := opentracing.StartSpanFromContext(ctx, "daemon-archiveDf")
+	span, _ := opentracing.StartSpanFromContext(ctx, "daemon-archiveDf")
```

Comment:
can we sink the ctx instead? i.e.,

```
_ = ctx
```

Overwriting the ctx is an important pattern to preserve i think